overlord/snapstate/handlers.go

```diff
@@ -899,6 +899,8 @@ func (m *SnapManager) doPreDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 // the user and then triggers an auto-refresh.
 func asyncRefreshOnSnapClose(st *state.State, refreshInfo *userclient.PendingSnapRefreshInfo) error {
 	var monitoredSnaps map[string]bool
+	st.Lock()
+	defer st.Unlock()
```

Comment:
I'm confused because Cached should panic without the lock? how was the code working before?  cc @MiguelPires

Reply:
I checked the code, and it worked because all the places where that function is called had a lock in place before.

Reply:
Anyway, the true race condition was in the goroutine where the entry was removed.

Reply:
Mmm... but now that I think... this would produce a lock...

Reply:
The locks are obtained in the handlers so this cause a deadlock

Reply:
Yes... Fixing it, and adding a comment in the function.

Reply:
Fixed.